exercises/practice/resistor-color/resistor-color_test.sql

```diff
@@ -0,0 +1,47 @@
+-- Create database:
+.read ./create_fixture.sql
+
+-- Read user student solution and save any output as markdown in user_output.md:
+.mode markdown
+.output user_output.md
+.read ./resistor-color.sql
+.output
+
+-- Create a clean testing environment:
+.read ./create_test_table.sql
+
+-- Comparison of user input and the tests updates the status for each test:
+UPDATE "test_color_code"
+SET status = 'pass'
+FROM (SELECT color, result FROM color_code) AS l
+WHERE (l.color, l.result) = (test_color_code.color, test_color_code.result);
+-- Update message for failed tests to give helpful information:
+UPDATE test_color_code
+SET message = 'Result for ' || l.color || ' is "' || l.result || '", but should be: "' || test_color_code.result || '"'
+FROM (SELECT color, result FROM color_code) AS l
+WHERE l.color = test_color_code.color AND test_color_code.status = 'fail';
+
+-- Comparison of user input and the tests updates the status for each test:
+UPDATE test_colors
+SET status = 'pass'
+FROM (SELECT GROUP_CONCAT("color" ORDER BY "index") as "result" FROM "colors") AS l
```

Comment:
Can we structure this exercise in such a way that the student has to come up with that sub-select by themselves? Perhaps we give them the `colors` table complete with index values, and demand they populate a column in some other result table with the comma separated colours in the right order.

Reply:
I'm a bit hesitant to wander too far from the problem spec here.

Would that be testing that they can use a `GROUP_CONCAT`? Is that related to the problem spec? Is that more of a concept exercise?

Should we just skip this test?

Reply:
Skipped.